codex-cli/src/utils/agent/agent-loop.ts

```diff
@@ -592,22 +633,37 @@ export class AgentLoop {
               errCtx.type === "rate_limit_exceeded" ||
               /rate limit/i.test(errCtx.message ?? "");
             if (isRateLimit) {
-              if (attempt < MAX_RETRIES) {
-                // Exponential backoff: base wait * 2^(attempt-1), or use suggested retry time
-                // if provided.
-                let delayMs = RATE_LIMIT_RETRY_WAIT_MS * 2 ** (attempt - 1);
+              if (attempt < maxRetries) {
+                // Get rate limit config or use defaults
+                const initialRetryDelayMs =
+                  this.config.rateLimits?.initialRetryDelayMs ??
+                  DEFAULT_RATE_LIMIT_INITIAL_RETRY_DELAY_MS;
+                const maxRetryDelayMs =
+                  this.config.rateLimits?.maxRetryDelayMs ??
+                  DEFAULT_RATE_LIMIT_MAX_RETRY_DELAY_MS;
+                const jitterFactor =
+                  this.config.rateLimits?.jitterFactor ??
+                  DEFAULT_RATE_LIMIT_JITTER_FACTOR;
 
                 // Parse suggested retry time from error message, e.g., "Please try again in 1.3s"
+                let suggestedDelayMs: number | undefined;
                 const msg = errCtx?.message ?? "";
                 const m = /retry again in ([\d.]+)s/i.exec(msg);
```

Comment:
is there not a retry-after header on the 429 response?

Reply:
^ +1

But _if_ string parsing is here to stay for this iteration, just wanted to call out that the rate-limiting error messages being shared have the substring `Please try again in`, not `retry again in`.